Library/Homebrew/cmd/shellenv.sh

```diff
@@ -9,30 +9,56 @@
-    fish|-fish)
+    fish | -fish)
       echo "set -gx HOMEBREW_PREFIX \"${HOMEBREW_PREFIX}\";"
       echo "set -gx HOMEBREW_CELLAR \"${HOMEBREW_CELLAR}\";"
       echo "set -gx HOMEBREW_REPOSITORY \"${HOMEBREW_REPOSITORY}\";"
-      echo "set -q PATH; or set PATH ''; set -gx PATH \"${HOMEBREW_PREFIX}/bin\" \"${HOMEBREW_PREFIX}/sbin\" \$PATH;"
-      echo "set -q MANPATH; or set MANPATH ''; set -gx MANPATH \"${HOMEBREW_PREFIX}/share/man\" \$MANPATH;"
-      echo "set -q INFOPATH; or set INFOPATH ''; set -gx INFOPATH \"${HOMEBREW_PREFIX}/share/info\" \$INFOPATH;"
+      ! contains ':' HOMEBREW_ENV_BACKUP_PATH "${HOMEBREW_PREFIX}/bin" "${HOMEBREW_PREFIX}/sbin" &&
+        echo "set -q PATH; or set PATH ''; set -gx PATH \"${HOMEBREW_PREFIX}/bin\" \"${HOMEBREW_PREFIX}/sbin\" \$PATH;"
+      ! contains ':' HOMEBREW_ENV_BACKUP_MANPATH "${HOMEBREW_PREFIX}/share/man" &&
+        echo "set -q MANPATH; or set MANPATH ''; set -gx MANPATH \"${HOMEBREW_PREFIX}/share/man\" \$MANPATH;"
+      ! contains ':' HOMEBREW_ENV_BACKUP_INFOPATH "${HOMEBREW_PREFIX}/share/info" &&
+        echo "set -q INFOPATH; or set INFOPATH ''; set -gx INFOPATH \"${HOMEBREW_PREFIX}/share/info\" \$INFOPATH;"
       ;;
-    csh|-csh|tcsh|-tcsh)
+    csh | -csh | tcsh | -tcsh)
       echo "setenv HOMEBREW_PREFIX ${HOMEBREW_PREFIX};"
       echo "setenv HOMEBREW_CELLAR ${HOMEBREW_CELLAR};"
       echo "setenv HOMEBREW_REPOSITORY ${HOMEBREW_REPOSITORY};"
-      echo "setenv PATH ${HOMEBREW_PREFIX}/bin:${HOMEBREW_PREFIX}/sbin:\$PATH;"
-      echo "setenv MANPATH ${HOMEBREW_PREFIX}/share/man\`[ \${?MANPATH} == 1 ] && echo \":\${MANPATH}\"\`:;"
-      echo "setenv INFOPATH ${HOMEBREW_PREFIX}/share/info\`[ \${?INFOPATH} == 1 ] && echo \":\${INFOPATH}\"\`;"
+      ! contains ':' HOMEBREW_ENV_BACKUP_PATH "${HOMEBREW_PREFIX}/bin" "${HOMEBREW_PREFIX}/sbin" &&
+        echo "setenv PATH ${HOMEBREW_PREFIX}/bin:${HOMEBREW_PREFIX}/sbin:\$PATH;"
+      ! contains ':' HOMEBREW_ENV_BACKUP_MANPATH "${HOMEBREW_PREFIX}/share/man" &&
+        echo "setenv MANPATH ${HOMEBREW_PREFIX}/share/man\`[ \${?MANPATH} == 1 ] && echo \":\${MANPATH}\"\`:;"
+      ! contains ':' HOMEBREW_ENV_BACKUP_INFOPATH "${HOMEBREW_PREFIX}/share/info" &&
+        echo "setenv INFOPATH ${HOMEBREW_PREFIX}/share/info\`[ \${?INFOPATH} == 1 ] && echo \":\${INFOPATH}\"\`;"
       ;;
     *)
       echo "export HOMEBREW_PREFIX=\"${HOMEBREW_PREFIX}\";"
       echo "export HOMEBREW_CELLAR=\"${HOMEBREW_CELLAR}\";"
       echo "export HOMEBREW_REPOSITORY=\"${HOMEBREW_REPOSITORY}\";"
-      echo "export PATH=\"${HOMEBREW_PREFIX}/bin:${HOMEBREW_PREFIX}/sbin\${PATH+:\$PATH}\";"
-      echo "export MANPATH=\"${HOMEBREW_PREFIX}/share/man\${MANPATH+:\$MANPATH}:\";"
-      echo "export INFOPATH=\"${HOMEBREW_PREFIX}/share/info:\${INFOPATH:-}\";"
+      ! contains ':' HOMEBREW_ENV_BACKUP_PATH "${HOMEBREW_PREFIX}/bin" "${HOMEBREW_PREFIX}/sbin" &&
+        echo "export PATH=\"${HOMEBREW_PREFIX}/bin:${HOMEBREW_PREFIX}/sbin\${PATH+:\$PATH}\";"
```

Comment:
So does this e.g. only output the `export` command if the `PATH` already contains both `bin` and `sbin`?

Reply:
> So does this e.g. only output the export command if the `PATH` already contains both `bin` and `sbin`?

Yes. If user add only one of them e.g. `export PATH="brew/bin:$PATH"` by themselves, `brew shellenv` will output the export command of `PATH`.